docs/building-modules/testing.md

```diff
@@ -0,0 +1,63 @@
+<!--
+order: 15
+-->
+
+# Module & App Testing
+
+The Cosmos SDK contains different types of [tests](https://martinfowler.com/articles/practical-test-pyramid.html).
+These tests have different goals and are used at different stages of the development cycle.
+We advice, as a general rule, to use tests at all stages of the development cycle.
+It is adviced, as a chain developer, to test your application and modules in a similar way than the SDK.
+
+The rationale behind testing can be found in [ADR-59](https://docs.cosmos.network/main/architecture/adr-059-test-scopes.html).
+
+## Unit Tests
+
+Unit tests are the lowest test category of the [test pyramid](https://martinfowler.com/articles/practical-test-pyramid.html).
```

Comment:
can we link to a tutorial on the internet about using mocks

Reply:
I think the link I keep linking explains everything about testing: https://martinfowler.com/articles/practical-test-pyramid.html. They talk about mocks too. I do agree that this should look more like a walkthrough. Updating!

Reply:
it doesn't have a link to how to write go mocks, this is generalised testing